core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -115,67 +117,214 @@ object LogConfig {
     "standard compression codecs ('gzip', 'snappy', lz4). It additionally accepts 'uncompressed' which is equivalent to " +
     "no compression; and 'producer' which means retain the original compression codec set by the producer."
   val PreAllocateEnableDoc ="Should pre allocate file when create new segment?"
-  val MessageFormatVersionDoc = KafkaConfig.LogMessageFormatVersionDoc
-  val MessageTimestampTypeDoc = KafkaConfig.LogMessageTimestampTypeDoc
-  val MessageTimestampDifferenceMaxMsDoc = KafkaConfig.LogMessageTimestampDifferenceMaxMsDoc
+  val MessageFormatVersionDoc = "Specify the message format version the broker will use to append messages to the logs. The value should be a valid ApiVersion. " +
+    "Some examples are: 0.8.2, 0.9.0.0, 0.10.0, check ApiVersion for more details. By setting a particular message format version, the " +
+    "user is certifying that all the existing messages on disk are smaller or equal than the specified version. Setting this value incorrectly " +
+    "will cause consumers with older versions to break as they will receive messages with a format that they don't understand."
+  val MessageTimestampTypeDoc = "Define whether the timestamp in the message is message create time or log append time. The value should be either " +
+    "`CreateTime` or `LogAppendTime`"
+  val MessageTimestampDifferenceMaxMsDoc = "The maximum difference allowed between the timestamp when a broker receives " +
+    "a message and the timestamp specified in the message. If message.timestamp.type=CreateTime, a message will be rejected " +
+    "if the difference in timestamp exceeds this threshold. This configuration is ignored if message.timestamp.type=LogAppendTime."  
+
+  private class LogConfigDef extends ConfigDef {
+
+    private final val serverDefaultConfigNames = mutable.Map[String, String]()
+
+    def define(name: String, defType: ConfigDef.Type, defaultValue: Object, validator: ConfigDef.Validator,
+               importance: ConfigDef.Importance, doc: String, serverDefaultConfigName: String) = {
+      super.define(name, defType, defaultValue, validator, importance, doc)
+      serverDefaultConfigNames.put(name, serverDefaultConfigName)
+      this
+    }
+
+    def define(name: String, defType: ConfigDef.Type, defaultValue: Object, importance: ConfigDef.Importance,
+               documentation: String, serverDefaultConfigName: String) = {
+      super.define(name, defType, defaultValue, importance, documentation)
+      serverDefaultConfigNames.put(name, serverDefaultConfigName)
+      this
+    }
+
+    def define(name: String, defType: ConfigDef.Type, importance: ConfigDef.Importance, documentation: String,
+               serverDefaultConfigName: String) = {
+      super.define(name, defType, importance, documentation)
+      serverDefaultConfigNames.put(name, serverDefaultConfigName)
+      this
+    }
+
+    override def toHtmlTable() = {
```

Comment:
I'm a bit hesitant about this override since it is 90% the same as the base class. But of course the server default property is definitely useful. The structure is exactly identical to `ConfigDef`'s version except the server default is inserted before the importance. It's definitely a bit awkward, but would it be better to just make the base class version extensible? It's not great since you need two overrides (extra headers and extra row fields), but it'd avoid a bunch of duplication. Also, https://github.com/apache/kafka/pull/1696 is admittedly for rst, but is extending the output with some fields added to ConfigDef to support richer config info. I think it might be better to try to hook into existing config output so we don't end up with diverging output formats.

Thoughts? Somehow I suspect @ijuma might have an opinion here (and will probably come up with a cleaner solution than I proposed).